src/test/scala/chiselTests/LTLSpec.scala

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.ltl._
+import chisel3.testers.BasicTester
+import _root_.circt.stage.ChiselStage
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import Sequence._
+
+class LTLSpec extends AnyFlatSpec with Matchers {
+  it should "allow booleans to be used as sequences" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new RawModule {
+      val a = IO(Input(Bool()))
+      Sequence.delay(a, 42)
+    })
+    chirrtl should include("intmodule LTLDelayIntrinsic_42_0 :")
+    chirrtl should include("input in : UInt<1>")
+    chirrtl should include("output out : UInt<1>")
+    chirrtl should include("intrinsic = circt_ltl_delay")
+    chirrtl should include("parameter delay = 42")
+    chirrtl should include("parameter length = 0")
+
+    chirrtl should include("inst ltl_delay of LTLDelayIntrinsic_42_0")
+    chirrtl should include("input a : UInt<1>")
+    chirrtl should include("ltl_delay.in <= a")
+  }
+
+  it should "allow booleans to be used as properties" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new RawModule {
+      val a = IO(Input(Bool()))
+      Property.eventually(a)
+    })
+    chirrtl should include("intmodule LTLEventuallyIntrinsic :")
+    chirrtl should include("input in : UInt<1>")
+    chirrtl should include("output out : UInt<1>")
+    chirrtl should include("intrinsic = circt_ltl_eventually")
+
+    chirrtl should include("inst ltl_eventually of LTLEventuallyIntrinsic")
+    chirrtl should include("input a : UInt<1>")
+    chirrtl should include("ltl_eventually.in <= a")
+  }
+
+  it should "support sequence delay operations" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new RawModule {
+      val a, b, c = IO(Input(Bool()))
+      val s0 = a.delay(1)
+      val s1 = b.delayRange(2, 4)
+      val s2 = c.delayAtLeast(5)
+      val s3 = a ### b
+      val s4 = a ##* b
+      val s5 = a ##+ b
+      require(s0.isInstanceOf[Sequence])
+      require(s1.isInstanceOf[Sequence])
+      require(s2.isInstanceOf[Sequence])
+      require(s3.isInstanceOf[Sequence])
+      require(s4.isInstanceOf[Sequence])
+      require(s5.isInstanceOf[Sequence])
```

Comment:
```suggestion
      val s0: Sequence = a.delay(1)
      val s1: Sequence = b.delayRange(2, 4)
      val s2: Sequence = c.delayAtLeast(5)
      val s3: Sequence = a ### b
      val s4: Sequence = a ##* b
      val s5: Sequence = a ##+ b
```
It's not really clear to me what these `requires` are trying to check, but if you're just trying to make sure the correct methods are being called, I think you'd be better of just with a static type annotation than the ugly runtime check.

Reply:
Same suggestion for all `requires` below.

Reply:
I was trying to be clever and not influence type inference with a static type annotation :grimacing: I'll drop the `require`s :+1: